lib/cfnguardian/stacks/main.rb

```diff
@@ -22,12 +22,42 @@ def build_template(stacks,checks,topics,maintenance_groups,ssm_parameters)
           parameter.Default sns
           parameters[name] = Ref(name)
         end
-        
+
+        if !maintenance_groups.empty?
+          add_lambda(CfnGuardian::Models::MaintenanceGroupCheck.new(maintenance_groups))
+        end
+
         maintenance_groups.each do |group|
-          topic = @template.SNS_Topic(group)
-          topic.TopicName group
+
+          group_name = "#{group[0]}MaintenanceGroup"
+
+          topic = @template.SNS_Topic(group_name)
+          topic.TopicName group_name
           topic.Tags([{ Key: 'Environment', Value: 'guardian' }])
-          parameters[group] = Ref(group)
+          parameters[group_name] = Ref(group_name)
+
+          if group[1].key?('Schedules')
+            enable_cron = group[1]['Schedules']['Enable']
+            disable_cron = group[1]['Schedules']['Disable']
+
+            event = @template.Events_Rule("#{group_name}EnableEvent")
+            event.Name "#{group_name}EnableEvent"
+            event.ScheduleExpression "cron(#{enable_cron})"
+            event.Targets([{ 
+              Arn: FnGetAtt('MaintenanceGroupCheckFunction', 'Arn'), 
+              Id: "#{group_name}EnableTarget", 
+              Input: "{\"action\":\"enable_alarms\",\"maintenance_group\":\"#{group_name}\"}"
```

Comment:
you could create a hash here and convert it to json which would make it a bit easier to read
```rb
{action:"enable_alarms", maintenance_group: group_name}.to_json
```